client/db/bolt/db.go

```diff
@@ -756,6 +819,55 @@ func (bp *bucketPutter) err() error {
 	return bp.putErr
 }
 
+// keyTimePair is used to build an on-the-fly time-sorted index.
+type keyTimePair struct {
+	k []byte
+	t uint64
+}
+
+// timeIndexNewest is a struct used to build an index of sorted keyTimePairs.
+// The index can have a maximum capacity. If the capacity is set to zero, the
+// index size is unlimited.
+type timeIndexNewest struct {
+	pairs []*keyTimePair
+	cap   int
+}
+
+// Create a new *timeIndexNewest, with the specified capacity.
+func newTimeIndexNewest(n int) *timeIndexNewest {
+	return &timeIndexNewest{
+		pairs: make([]*keyTimePair, 0, n),
+		cap:   n,
+	}
+}
+
+// Conditionally add a time-key pair to the index. The pair will only be added
+// if the timeIndexNewest is under capacity and the time t is larger than the
+// oldest pair's time.
+func (idx *timeIndexNewest) add(t uint64, k []byte) {
+	count := len(idx.pairs)
+	if idx.cap == 0 || count < idx.cap { // count = 0 always evals true here.
```

Comment:
Maybe I'm being dense, but I don't get the comment `// count = 0 always evals true here.`.

Reply:
I was just reminding myself that `count = 0` is handled by the first block since I'm not checking  that condition explicitly. It's obviously more harm than help though. Will erase.